### PR TITLE
response header sometimes doesn't give a content-length.

### PR DESCRIPTION
**Pull request by candh**
we want to quit with an error before its passed on to parseInt.

I gave annie a openload.co link, it tried to download it directly and gave me this error

```bash 
$ annie https://openload.co/f/kKNtu5AZ3mE/Monty_Python_e_il_Sacro_Graal.mp4

annie doesn't support this URL right now, but it will try to download it directly
Downloading https://openload.co/f/kKNtu5AZ3mE/Monty_Python_e_il_Sacro_Graal.mp4 error:
strconv.ParseInt: parsing "": invalid syntax
```

My mistake, i noticed the "download" button on the site afterwards. Content-Length doesn't get set on openload.co links. So, we print a (somewhat) nicer error and quit.

### CODE REVIEW

**Comment by candh** [comment]
hmm, why did the test fail? :cry:

File: request/request.go [hunk 1]
```diff
@@ -175,6 +175,11 @@ func Size(url, refer string) (int64, error) {
 		return 0, err
 	}
 	s := h.Get("Content-Length")
+	if s == "" {
+		var err error
+		err = fmt.Errorf("Content-Length is not present")
+		return 0, err
```
Comment (iawia002) [inline request/request.go]: return `0, errors.New("Content-Length is not present")` would be better

Comment (candh) [reply request/request.go]: absolutely, yeah!

**Comment by candh** [comment]
yep, its good now. i think...

**Pull request merged**